icu4c/source/common/uloc_tag.cpp

```diff
@@ -2049,7 +2049,10 @@ ultag_parse(const char* tag, int32_t tagLen, int32_t* parsedLen, UErrorCode* sta
         *status = U_MEMORY_ALLOCATION_ERROR;
         return NULL;
     }
-    uprv_memcpy(tagBuf, tag, tagLen);
+    
+    if (tagLen > 0 && tag != nullptr) {
```

Comment:
```suggestion
    if (tagLen > 0) {
```
This should do. If we still get a nullptr, we want to know.